Fabric.Authorization.Domain/Services/GroupService.cs

```diff
@@ -202,5 +202,20 @@ public class GroupService
 
             return await _groupStore.DeleteUserFromGroup(group, user);
         }
+
+        public async Task<IEnumerable<Group>> GetGroups(string name, string type)
+        {
+            if (!string.IsNullOrEmpty(type))
+            {
+                type = type.ToLower();
+                if (!type.Equals("custom") && !type.Equals("directory"))
+                {
+                    throw new BadRequestException<Group>("Invalid type provided. If provided valid values are custom or directory");
```

Comment:
Since this is in the domain layer, should we change this to an InvaildOperationException? "Bad Request" is an HTTP type of concept which I don't think we should have in the domain layer. The Web layer can catch the InvalidOperationException and create a bad request response.